src/Kiota.Builder/Refiners/JavaRefiner.cs

```diff
@@ -298,9 +299,72 @@ public override Task Refine(CodeNamespace generatedCode, CancellationToken cance
 
         CrawlTree(currentElement, InsertOverrideMethodForRequestExecutorsAndBuildersAndConstructors);
     }
+    private static void RemoveClassNamePrefixFromNestedClasses(CodeElement currentElement) {
+        if(currentElement is CodeClass currentClass) {
```

Comment:
```suggestion
        if(currentElement is CodeClass currentClass && currentClass.IsOfKind(CodeClassKind.RequestBuilder)) {
```
To save some unnecessary work, we could probably first check if its a requestBuilder as `RequestConfiguration` instances are exclusively nested inside them at the momement.

Reply:
good catch there, I agree.